protocol/protocol_test.go

```diff
@@ -0,0 +1,212 @@
+package protocol
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/jackc/pgx/pgproto3"
+	pgstories "github.com/panoplyio/pg-stories"
+	"io"
+	"net"
+	"testing"
+	"time"
+)
+
+func TestProtocol_StartUp(t *testing.T) {
+
+	t.Run("supported protocol version", func(t *testing.T) {
+
+		buf := bytes.Buffer{}
+		comm := bufio.NewReadWriter(bufio.NewReader(&buf), bufio.NewWriter(&buf))
+		p := &Protocol{W: comm, R: comm}
+
+		if _, err := comm.Write([]byte{0, 0, 0, 8, 0, 3, 0, 0, 0, 0, 0, 0}); err != nil {
```

Comment:
Can you rewrite the message using a more human-friendly layout? For example, see `auth_test.go`:
```go
	passwordRequest := protocol.Message{
		'R',
		0, 0, 0, 8, // length
		0, 0, 0, 3, // clear text auth type
	}
```
This comment also applies to similar tests.